tonic/src/codec/compression.rs

```diff
@@ -104,13 +137,19 @@ impl CompressionEncoding {
         let header_value = map.get(ACCEPT_ENCODING_HEADER)?;
         let header_value_str = header_value.to_str().ok()?;
 
-        split_by_comma(header_value_str).find_map(|value| match value {
-            #[cfg(feature = "gzip")]
-            "gzip" => Some(CompressionEncoding::Gzip),
-            #[cfg(feature = "zstd")]
-            "zstd" => Some(CompressionEncoding::Zstd),
-            _ => None,
+        // Get the highest supported encoding and its priority
+        split_by_comma(header_value_str).filter_map(|value| {
+            match value {
+                #[cfg(feature = "gzip")]
+                "gzip" => Some(CompressionEncoding::Gzip),
+                #[cfg(feature = "zstd")]
+                "zstd" => Some(CompressionEncoding::Zstd),
+                _ => None,
+            }.map(|encoding| (encoding, enabled_encodings.priority(encoding)))
         })
+            .take(COMPRESSION_ENCODINGS_LENGTH)
```

Comment:
This is probably not correct. (see the line 79)

Reply:
Ah yeah, will add a +1